src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -989,32 +1029,21 @@ final class LazyList[+A] private(private[this] var lazyState: () => LazyList.Sta
   * @define coll lazy list
   * @define Coll `LazyList`
   */
-@SerialVersionUID(3L)
+@SerialVersionUID(4L)
 object LazyList extends SeqFactory[LazyList] {
-  // Eagerly evaluate cached empty instance
-  private[this] val _empty = newLL(State.Empty).force
-
-  private sealed trait State[+A] extends Serializable {
-    def head: A
-    def tail: LazyList[A]
-  }
+  private object Uninitialized extends Serializable
+  private object MidEvaluation
 
-  private object State {
-    @SerialVersionUID(3L)
-    object Empty extends State[Nothing] {
-      def head: Nothing = throw new NoSuchElementException("head of empty lazy list")
-      def tail: LazyList[Nothing] = throw new UnsupportedOperationException("tail of empty lazy list")
-    }
-
-    @SerialVersionUID(3L)
-    final class Cons[A](val head: A, val tail: LazyList[A]) extends State[A]
-  }
+  private val EmptyFun = () => null
+  private val Empty = new LazyList(EmptyFun)
+  // Eagerly evaluate cached empty instance
+  locally(Empty.force)
 
   /** Creates a new LazyList. */
-  @inline private def newLL[A](state: => State[A]): LazyList[A] = new LazyList[A](() => state)
+  @inline private def newLL[A](state: => LazyList[A]): LazyList[A] = new LazyList[A](() => state)
 
   /** Creates a new State.Cons. */
-  @inline private def sCons[A](hd: A, tl: LazyList[A]): State[A] = new State.Cons[A](hd, tl)
+  @inline private def sCons[A](hd: A, tl: LazyList[A]): LazyList[A] = new LazyList[A](hd, tl)
```

Comment:
```suggestion
  @inline private def eagerCons[A](hd: A, tl: LazyList[A]): LazyList[A] = new LazyList[A](hd, tl)
```

comment and every time this method is called would also need updating, obviously